### PR TITLE
Check dsn in config before throw error

### PR DESCRIPTION
**Pull request by jannyHou**
Connect to strongloop/loopback-connector-db2#34

### CODE REVIEW

**Comment by jannyHou** [comment]
@qpresley This is a fix for issue https://github.com/strongloop/loopback-connector-db2/issues/34
Please review:) Thanks

File: test/db2.connection.test.js [hunk 1]
```diff
@@ -24,6 +24,19 @@ describe('testConnection', function() {
...
+  it('should pass when valid DSN overrides empty settings', function(done) {
+    var dsn = generateDSN(config);
+    var dbConfig = {
+      dsn: dsn,
+    };
+
+    var db = new DataSource(require('../'), dbConfig);
+    db.ping(function(err) {
+      assert(!err, 'Should connect without err.');
+      done(err);
```
Comment (superkhau) [inline test/db2.connection.test.js]: Do you need done(err)? Looks like it can just be done();

Comment (jannyHou) [inline test/db2.connection.test.js]: Sounds good. Fixed by `done()`
`done(err)` will affect the mocha error throw.

**Comment by qpresley** [comment]
LGTM

**Pull request merged**